arlo-client/src/components/Audit/useSetupMenuItems/index.tsx

```diff
@@ -22,20 +26,50 @@ function useSetupMenuItems(
   const [reviewLaunch, setReviewLaunch] = useState<ISidebarMenuItem['state']>(
     'live'
   )
+  const setContests = useCallback(
+    (s: ISidebarMenuItem['state']) => {
+      setTargetContests(s)
+      setOpportunisticContests(s)
+    },
+    [setTargetContests, setOpportunisticContests]
```

Comment:
@jonahkagan The premise is that every time a functional component renders, it calls everything in it again, including `const` declarations. Therefore, each rerender, every function is created all over again from scratch. Thus, it's not the result you want cached, it's the function itself. This is initially just a performance thing, but when you incorporate hooks, particularly `useEffect`, it becomes critical. An effect hook is rerun every time any of its dependencies changes (specified in the second argument in the array there). So if any of those functions are re-created, it will re-call the effect, which almost inevitably leads to infinite loops. Therefore, putting functions inside of `useCallback` will remove this factor. In fact, react has a built in linter that warns when dependencies aren't added, or if there are extra ones that shouldn't be added. I initially left out the state setters, because of that same documentation you referenced, but the linter required I put them in. * shrug * My hypothesis is that it might be due to being inside of a custom hook? Anyway, I'm playing it safe and following the linter's instructions, haha. Thanks for checking though!